client_encryption/json_path_utils.py

```diff
@@ -42,7 +42,12 @@ def update_node(tree, path, node_str):
         except json.JSONDecodeError:
             node_json = node_str
 
-        if to_set in current_node and type(current_node[to_set]) is dict and type(node_json) is dict:
+        if type(current_node) is list:
```

Comment:
You could add more test cases for this file as well since there is a decline in the code coverage:
https://sonarcloud.io/component_measures?metric=new_coverage&selected=Mastercard_client-encryption-python%3Aclient_encryption%2Fjson_path_utils.py&view=list&branch=fix_array_payload&id=Mastercard_client-encryption-python